gateway/src/node/statistics/mod.rs

```diff
@@ -1,4 +1,63 @@
 // Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
 // SPDX-License-Identifier: GPL-3.0-only
 
-pub mod collector;
+use futures::{channel::mpsc, StreamExt};
+use nym_node_http_api::state::metrics::SharedSessionStats;
+use nym_statistics_common::events::{StatsEvent, StatsEventReceiver, StatsEventSender};
+use nym_task::TaskClient;
+use sessions::SessionStatsHandler;
+use std::time::Duration;
+use time::OffsetDateTime;
+use tracing::trace;
+
+pub mod sessions;
+
+const STATISTICS_UPDATE_TIMER_INTERVAL: Duration = Duration::from_secs(3600); //update timer, no need to check everytime
+
+pub(crate) struct GatewayStatisticsCollector {
+    stats_event_rx: StatsEventReceiver,
+    session_stats: SessionStatsHandler,
+    //here go additionnal stats handler
+}
+
+impl GatewayStatisticsCollector {
+    pub fn new(
+        shared_session_stats: SharedSessionStats,
+    ) -> (GatewayStatisticsCollector, StatsEventSender) {
+        let (stats_event_tx, stats_event_rx) = mpsc::unbounded();
+        let collector = GatewayStatisticsCollector {
+            stats_event_rx,
+            session_stats: SessionStatsHandler::new(shared_session_stats),
+        };
+        (collector, stats_event_tx)
+    }
+
+    async fn update_shared_state(&mut self, update_time: OffsetDateTime) {
+        self.session_stats.update_shared_state(update_time).await;
+        //here go additionnal stats handler update
+    }
+
+    pub async fn run(&mut self, mut shutdown: TaskClient) {
+        let mut update_interval = tokio::time::interval(STATISTICS_UPDATE_TIMER_INTERVAL);
+        while !shutdown.is_shutdown() {
+            tokio::select! {
```

Comment:
Consider handling the None case for `stats_event_rx.next()`, conveniently by having a `else => ` branch in the select

Reply:
I can't think of anything we want to do if that's the case. Sure we won't update the internal state, but we still need to update the shared object for the described API, so we can't stop.